apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/model/oms/Util.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.etterlatte.brev.model.oms
+
+import no.nav.etterlatte.brev.model.OmstillingsstoenadBeregningsperiode
+import no.nav.etterlatte.libs.common.behandling.DetaljertBehandling
+import java.time.LocalDate
+
+fun utledOpphoer(
+    behandling: DetaljertBehandling,
+    beregningsperioder: List<OmstillingsstoenadBeregningsperiode>,
+): LocalDate? {
+    val sisteBeregningsperiode =
+        beregningsperioder
+            .filter {
+                it.datoFOM.year == beregningsperioder.first().datoFOM.year
+            }.maxBy { it.datoFOM }
+    val sisteBeregningsperiodeNesteAar =
+        beregningsperioder
+            .filter {
+                it.datoFOM.year == beregningsperioder.first().datoFOM.year + 1
+            }.maxByOrNull { it.datoFOM }
+
+    // Hvis antall innvilga måneder er overstyrt under beregning skal "forventa" opphørsdato vises selv uten opphørFom
+    val forventaOpphoersDato =
+        when (val opphoer = behandling.opphoerFraOgMed) {
+            null -> {
+                if (sisteBeregningsperiode.erOverstyrtInnvilgaMaaneder) {
+                    val foersteFom = beregningsperioder.first().datoFOM
+                    foersteFom.plusMonths(sisteBeregningsperiode.innvilgaMaaneder.toLong())
+                } else if (sisteBeregningsperiodeNesteAar != null && sisteBeregningsperiodeNesteAar.erOverstyrtInnvilgaMaaneder) {
+                    LocalDate
+                        .of(sisteBeregningsperiodeNesteAar.datoFOM.year, 1, 1)
+                        .plusMonths(sisteBeregningsperiodeNesteAar.innvilgaMaaneder.toLong())
+                } else {
```

Comment:
Mangler et case for aldersovergang 67 år da

Reply:
Skal det håndteres likt som i avkorting må opphørsdato hentes fra grunnlag og sjekkes her

Reply:
Jepp, vi trenger nok å endre denne til å se på innvilgede måneder og kompensere for innvilgelse